src/libMessage/Messenger.cpp

```diff
@@ -360,3 +360,836 @@ bool Messenger::GetNodeFinalBlock(const vector<unsigned char>& src,
+    {
+        Peer seedPeer;
+        ProtobufByteArrayToSerializable(result.candidateseeds(i), seedPeer);
+        candidateSeeds.emplace_back(seedPeer);
+    }
+
+    return true;
+}
+
+bool Messenger::SetLookupGetDSInfoFromSeed(vector<unsigned char>& dst,
+                                           const unsigned int offset,
+                                           const uint32_t listenPort)
+{
+    LOG_MARKER();
+
+    LookupGetDSInfoFromSeed result;
+
+    result.set_listenport(listenPort);
+
+    if (result.IsInitialized() == false)
+    {
+        LOG_GENERAL(WARNING, "LookupGetDSInfoFromSeed initialization failed.");
+        return false;
+    }
+
+    return SerializeToArray(result, dst, offset);
+}
+
+bool Messenger::GetLookupGetDSInfoFromSeed(const vector<unsigned char>& src,
+                                           const unsigned int offset,
+                                           uint32_t& listenPort)
+{
+    LOG_MARKER();
+
+    LookupGetDSInfoFromSeed result;
+
+    result.ParseFromArray(src.data() + offset, src.size() - offset);
+
+    if (result.IsInitialized() == false)
+    {
+        LOG_GENERAL(WARNING, "LookupGetDSInfoFromSeed initialization failed.");
+        return false;
+    }
+
+    listenPort = result.listenport();
+
+    return true;
+}
+
+bool Messenger::SetLookupSetDSInfoFromSeed(
+    vector<unsigned char>& dst, const unsigned int offset,
+    const vector<pair<PubKey, Peer>>& dsNodes)
+{
+    LOG_MARKER();
+
+    LookupSetDSInfoFromSeed result;
+
+    for (const auto& node : dsNodes)
+    {
+        LookupSetDSInfoFromSeed::DSNode* protodsnode = result.add_dsnodes();
+        SerializableToProtobufByteArray(node.first,
+                                        *protodsnode->mutable_pubkey());
+        SerializableToProtobufByteArray(node.second,
+                                        *protodsnode->mutable_peer());
+    }
+
+    if (result.IsInitialized() == false)
+    {
+        LOG_GENERAL(WARNING, "LookupSetDSInfoFromSeed initialization failed.");
+        return false;
+    }
+
+    return SerializeToArray(result, dst, offset);
+}
+
+bool Messenger::GetLookupSetDSInfoFromSeed(const vector<unsigned char>& src,
+                                           const unsigned int offset,
+                                           vector<pair<PubKey, Peer>>& dsNodes)
+{
+    LOG_MARKER();
+
+    LookupSetDSInfoFromSeed result;
+
+    result.ParseFromArray(src.data() + offset, src.size() - offset);
+
+    if (result.IsInitialized() == false)
+    {
+        LOG_GENERAL(WARNING, "LookupSetDSInfoFromSeed initialization failed.");
+        return false;
+    }
+
+    for (int i = 0; i < result.dsnodes_size(); i++)
+    {
+        PubKey pubkey;
+        Peer peer;
+
+        ProtobufByteArrayToSerializable(result.dsnodes(i).pubkey(), pubkey);
+        ProtobufByteArrayToSerializable(result.dsnodes(i).peer(), peer);
+        dsNodes.emplace_back(make_pair(pubkey, peer));
```

Comment:
The reason we encourage emplace_back is because there would be no temporary memory be used. That is, it's unnecessary to use make_pair here to improve some efficiency.

Reply:
Ok will fix